### PR TITLE
build/adm-storage

### PR DESCRIPTION
**Pull request by ryanbtp**
This is probably worth PRing for discussion, though I still lack a green build mostly due to making my life difficult with Hipster Silicon

### CODE REVIEW

File: adm/src/config.rs [hunk 1]
```diff
@@ -24,7 +24,8 @@ const DEFAULT_DATA_DIR: &str = "/var/lib/sawtooth";
 const DEFAULT_KEY_DIR: &str = "/etc/sawtooth/keys";
 const DEFAULT_POLICY_DIR: &str = "/etc/sawtooth/policy";
 
-const DEFAULT_BLOCKSTORE_FILENAME: &str = "block-00.lmdb";
+const DEFAULT_BLOCKSTORE_FILENAME_LMDB: &str = "block-00.lmdb";
+const DEFAULT_BLOCKSTORE_FILENAME_ROCKSDB: &str = "block-00.rocksdb";
```
Comment (ryanbtp) [inline adm/src/config.rs:28]: The admin tool appears to have no options for the blockstore name, so this is just a convention based on paths from sawtooth env vars?

File: database/src/rocks_db.rs [hunk 2]
```diff
@@ -70,24 +70,24 @@ impl Database for RocksDB {
     }
 
     fn flush(&self) {
-        debug!("Flushing rocks db at {:?}", self.path);
+        log::debug!("Flushing rocks db at {:?}", self.path);
```
Comment (ryanbtp) [inline database/src/rocks_db.rs:73]: Ok this is inconsistent, before I realised we were on 2018, will remove the namespace

File: database/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+version = "0.1.0"
+authors = ["sawtooth"]
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+[dependencies]
+cpython = "0.6"
+python3-sys = "0.6"
+lmdb-zero = ">=0.4.1"
+log = { version = "0.4", features = ["std"] }
+rocksdb = { version = ">=0.15.0", default-features = false, features = [ "lz4", "snappy", "zlib", "bzip2" ] }
```
Comment (scealiontach) [inline database/Cargo.toml:27]: need a newline here

File: database/src/error.rs [hunk 4]
```diff
@@ -52,7 +52,7 @@ impl std::error::Error for DatabaseError {
         }
     }
 
-    fn cause(&self) -> Option<&std::error::Error> {
+    fn cause(&self) -> Option<&dyn std::error::Error> {
```
Comment (scealiontach) [inline database/src/error.rs:55]: Why the dyn here? meaning I thought that was only for traits, and I thought Error was a concrete

**Review by scealiontach — CHANGES_REQUESTED** [review]
One trivial change requested, and just a question for my own edification.  Otherwise LGTM

**Pull request merged**